cardinal-sdk/src/commonTest/kotlin/com/icure/cardinal/sdk/crypto/KeylessPatientUserTest.kt

```diff
@@ -0,0 +1,314 @@
+					user = patientUser,
+					secretId = SecretIdUseOption.UseNone,
+					alternateRootDataOwnerReference = EntityReferenceInGroup(hcp.dataOwnerId)
+				)
+			)
+
+		}
+
+		"A keyless API should not allow the creation of encrypted data if no exchange data is provided" {
+			val patientApi = patientDetails.api(
+				specJob,
+				object : CryptoStrategies {
+					override suspend fun generateNewKeyForDataOwner(
+						self: DataOwnerWithType,
+						cryptoPrimitives: CryptoService,
+					): CryptoStrategies.KeyGenerationRequestResult {
+						return CryptoStrategies.KeyGenerationRequestResult.Keyless
+					}
+				},
+			)
+
+			val patientUser = patientApi.user.getCurrentUser()
+			val patient = patientApi.patient.encrypted.getPatient(patientUser.dataOwnerId).shouldNotBeNull()
+
+			runCatching {
+				createCalendarItem(patientApi, patient, patientUser)
+			}.isFailure shouldBe true
+		}
+
+		"A keyless API should allow creation of data and retrieval by delegator and delegate" {
+			val (patientApi, exchangeData) = getPatientApi()
+
+			// Arrange: initialize test data
+			val patientUser = patientApi.user.getCurrentUser()
+			val patient = patientApi.patient.encrypted.getPatient(patientUser.dataOwnerId).shouldNotBeNull()
+
+			val ci = createCalendarItem(patientApi, patient, patientUser)
+
+			// Patient retrieves the calendar item
+			val retrievedByPatient = patientApi.calendarItem.getCalendarItem(ci.id)!!
+			retrievedByPatient.title shouldBe ci.title
+			patientApi.calendarItem.decryptPatientIdOf(retrievedByPatient).map { it.entityId } shouldBe listOf(patient.id)
+
+			// HCP retrieves the calendar item
+			val retrievedByHcp = hcpApi.calendarItem.getCalendarItem(ci.id)!!
+			retrievedByHcp.title shouldBe ci.title
+			hcpApi.calendarItem.decryptPatientIdOf(retrievedByHcp).map { it.entityId } shouldBe listOf(patient.id)
+		}
+
+		"A keyless API should allow retrieval in new instance after exchange data injection" {
+			// Arrange: initialize test data
+			val (patientApi, exchangeData) = getPatientApi()
+
+			val patientUser = patientApi.user.getCurrentUser()
+			val patient = patientApi.patient.encrypted.getPatient(patientUser.dataOwnerId).shouldNotBeNull()
+
+			val ci = createCalendarItem(patientApi, patient, patientUser)
+
+			val retrievedByPatient = patientApi.calendarItem.getCalendarItem(ci.id)!!
+			retrievedByPatient.title shouldBe ci.title
+
+			// Simulate a new API instance (new session)
+			val newPatientApi = patientDetails.api(
+				specJob,
+				object : CryptoStrategies {
+					override suspend fun generateNewKeyForDataOwner(
+						self: DataOwnerWithType,
+						cryptoPrimitives: CryptoService,
+					): CryptoStrategies.KeyGenerationRequestResult {
+						return CryptoStrategies.KeyGenerationRequestResult.Keyless
+					}
+				},
+			)
+
+			// Before injecting exchange data, retrieval should fail
+			runCatching {
+				newPatientApi.calendarItem.getCalendarItem(ci.id)
+			}.isFailure shouldBe true
+
```

Comment:
Use `shouldThrow`. In this case if i'm not mistaken it should be a request exception and you should check for the status being 403